operations/host_spawn.go

```diff
@@ -792,3 +796,367 @@ func hostRunCommand() cli.Command {
+			},
+			cli.StringFlag{
+				Name:  joinFlagNames(remotePathFlagName, "r"),
+				Usage: "the remote directory/file (required)",
+			},
+			cli.BoolFlag{
+				Name:  remoteIsLocalFlagName,
+				Usage: "if set, both the source and destination filepaths are on the local machine (host does not need to be specified)",
+			},
+			cli.StringSliceFlag{
+				Name:  joinFlagNames(excludeFlagName, "x"),
+				Usage: "ignore syncing any files matched by the given pattern",
+			},
+			cli.BoolTFlag{
+				Name:  joinFlagNames(deleteFlagName, "d"),
+				Usage: "delete any files in the destination directory that are not present on the source directory (default: true)",
+			},
+			cli.BoolTFlag{
+				Name:  joinFlagNames(makeParentDirectoriesFlagName, "p"),
+				Usage: "create parent directories for the destination if they do not already exist (default: true)",
+			},
+			cli.DurationFlag{
+				Name:  joinFlagNames(timeoutFlagName, "t"),
+				Usage: "timeout for rsync command, e.g. '5m' = 5 minutes, '30s' = 30 seconds (if unset, there is no timeout)",
+			},
+			cli.BoolFlag{
+				Name:  joinFlagNames(pullFlagName),
+				Usage: "pull files from the remote host to the local host (default is to push files from the local host to the remote host)",
+			},
+			cli.BoolTFlag{
+				Name:  sanityChecksFlagName,
+				Usage: "perform basic sanity checks for common sources of error (ignored on dry runs) (default: true)",
+			},
+			cli.BoolFlag{
+				Name:  joinFlagNames(dryRunFlagName, "n"),
+				Usage: "show what would occur if executed, but do not actually execute",
+			},
+		),
+		Before: mergeBeforeFuncs(
+			func(c *cli.Context) error {
+				if c.Bool(remoteIsLocalFlagName) {
+					if c.String(hostFlagName) != "" {
+						return errors.New("should not specify a remote host ID when both filepaths are on the local machine")
+					}
+					return nil
+				}
+				return requireHostFlag(c)
+			},
+			requireStringFlag(localPathFlagName),
+			requireStringFlag(remotePathFlagName),
+		),
+		Action: func(c *cli.Context) error {
+			doSanityCheck := c.BoolT(sanityChecksFlagName)
+			localPath := c.String(localPathFlagName)
+			remotePath := c.String(remotePathFlagName)
+			pull := c.Bool(pullFlagName)
+			dryRun := c.Bool(dryRunFlagName)
+			remoteIsLocal := c.Bool(remoteIsLocalFlagName)
+
+			if strings.HasSuffix(localPath, string(os.PathSeparator)) && !strings.HasSuffix(remotePath, string(os.PathSeparator)) {
+				remotePath = remotePath + string(os.PathSeparator)
+			}
+			if strings.HasSuffix(remotePath, string(os.PathSeparator)) && !strings.HasSuffix(localPath, string(os.PathSeparator)) {
+				localPath = localPath + string(os.PathSeparator)
+			}
+
+			ctx, cancel := context.WithCancel(context.Background())
+			defer cancel()
+
+			var user, host string
+			var err error
+			if !remoteIsLocal {
+				hostID := c.String(hostFlagName)
+				user, host, err = getUserAndHostname(ctx, c.String(hostFlagName), c.Parent().Parent().String(confFlagName))
```

Comment:
should we also check the host is running?

Reply:
Done.